tornado/gen.py

```diff
@@ -711,9 +711,15 @@ def _contains_yieldpoint(children):
     and `multi_future`.
     """
     if isinstance(children, dict):
-        return any(isinstance(i, YieldPoint) for i in children.values())
+        YieldPoint_ = YieldPoint
+        for i in children.values():
+            if isinstance(i, YieldPoint_):
+                return True
```

Comment:
Aren't you missing the `return False`? If the value is a dic, and one value isn't an instance of `YieldPoint`, then it will check the other if to check if instance of list.. While the original code already returned False.